app/controllers/idv/in_person/address_controller.rb

```diff
@@ -73,7 +95,14 @@ def confirm_in_person_state_id_step_complete
       def confirm_in_person_address_step_needed
         return if pii_from_user && pii_from_user[:same_address_as_id] == 'false' &&
                   !pii_from_user.has_key?(:address1)
-        redirect_to idv_in_person_ssn_url
+        return if request.referer == idv_in_person_verify_info_url
+        redirect_to idv_in_person_verify_info_url
+      end
+
+      def confirm_ssn_step_needed
+        if pii_from_user&.has_key?(:address1) && !idv_session.ssn
+          redirect_to idv_in_person_ssn_url
+        end
```

Comment:
Why do you need this check here? I would think the SsnController would do this check if needed.

Reply:
i added it in to handle someone manually entering `in_person_proofing/address` on the ssn pg